src/components/nano/ModalSelectAddressToSignData.js

```diff
@@ -0,0 +1,171 @@
+import AddressList from '../../components/AddressList';
+import hathorLib from '@hathor/wallet-lib';
+import { NANO_UPDATE_ADDRESS_LIST_COUNT } from '../../constants';
+import { getGlobalWallet } from '../../modules/wallet';
+import PinInput from '../PinInput';
+
+/**
+ * Component that shows a modal to select an address
+ * to sign a nano contract signed data parameter
+ *
+ * @param props.onSuccess Method to be called when the sign succeeds
+ * @param props.onClose GlobalModal method added in all modals
+ *
+ * @memberof Components
+ */
+function ModalSelectAddressToSignData({ onClose, onSuccess }) {
+  const wallet = getGlobalWallet();
+
+  const addressToSignModalID = 'selectAddressToSignModal';
+
+  // {number} step The modal has 2 steps: step 0 is the address selection to sign the data,
+  //               step1 is the data and PIN input, so we can sign the data and close the modal
+  const [step, setStep] = useState(0);
+  // {Object} address Selected address to sign the data {'address': string, 'index': number}
+  const [address, setAddress] = useState(null);
+  // {string} in case there's an error in the sign process
+  const [error, setError] = useState('');
+
+  const formSignDataRef = useRef(null);
+  const pinRef = useRef(null);
+  const dataRef = useRef(null);
+
+  useEffect(() => {
+    $(`#${addressToSignModalID}`).modal('show');
+    $(`#${addressToSignModalID}`).on('hidden.bs.modal', (e) => {
+      setStep(0);
+      // We always need to call on close when using global context modal
+      // it will remove event listeners
+      onClose();
+    })
+  }, []);
+
+  /**
+   * Called when user selects one address, so we continue to next step
+   *
+   * @param {string} address Address clicked
+   */
+  const onAddressClick = async (address) => {
+    const addressIndex = await wallet.getAddressIndex(address);
+    setAddress({ address, index: addressIndex});
+    setStep(1);
+  }
+
+  /**
+   * Method called when user clicks the button to sign the data
+   *
+   * @param {Object} e Event emitted when user clicks the button
+   */
+  const signData = async (e) => {
+    e.preventDefault();
+
+    const isValid = formSignDataRef.current.checkValidity();
+    if (!isValid) {
+      formSignDataRef.current.classList.add('was-validated')
+      return;
+    }
+
+    const pin = pinRef.current.refs.pin.value;
+    // Incorrect PIN, show error message and do nothing else
+    if (!await wallet.checkPin(pin)) {
+      // TODO move side effects to saga
+      setError(t`Invalid PIN`);
+      return;
+    }
+
+    const data = dataRef.current.value;
+    const oracleData = hathorLib.nanoUtils.getOracleBuffer(address.address, wallet.getNetworkObject());
+    const nanoSerializer = new hathorLib.NanoContractSerializer();
+    const dataSerialized = nanoSerializer.serializeFromType(data, 'str');
+    // TODO getOracleInputData method should be able to receive the PIN as optional parameter as well
+    wallet.pinCode = pin;
```

Comment:
chore: We should clear the `pinCode` from the global wallet as soon as it's used by the `getOracleInputData()`. Am I right?

Reply:
Done https://github.com/HathorNetwork/hathor-wallet/pull/630/commits/6bfed308d04fd5cc61fe7e05c7e94fcb0b56bf2b